beacon/utils.py

```diff
@@ -153,3 +155,6 @@ def ERROR_MESSAGE(self):
             "<scheme>://<api-key>@<beacon-server-hostname>/<repository-id>"
             .format(self.input_dsn_string)
         )
+
+    def has_secure_scheme(self):
+        return self.scheme.lower() == 'https'
```

Comment:
Conversion to lowercase isn't needed here, because only lowercase scheme is accepted in `_set_scheme`.

Also, this could be a property of this class, rather than a method.

Reply:
But the `utils.DSN` class does not have any context about that (`_set_scheme`). I think this should be treated independently.